os/kernel/debug/sysdbg.c

```diff
@@ -421,53 +428,69 @@ static ssize_t sysdbg_write(FAR struct file *filep, FAR const char *buffer, size
 
 	sscanf(buffer, "%s %s", arg1, arg2);
 
-	if (strncmp("enable_monitor", arg1, 14) == 0) {
+	if (strncmp("help", arg1, 5) == 0) {
+		show_usage();
+		return 0;
+	}
+
+	switch (atoi(arg1)) {
+	case 1:		/*Enable monitor*/
 		sysdbg_monitor_enable();
-	} else if (strncmp("disable_monitor", arg1, 15) == 0) {
+		break;
+	case 2:		/*Disable monitor*/
 		sysdbg_monitor_disable();
-	} else if (strncmp("read", arg1, 4) == 0) {
+		break;
+	case 3:		/*Read /dev/sysdbg to view debug information*/
 		sysdbg_print();
-	} else if (strncmp(arg1, "max_task_count", 14) == 0) {
+		break;
+	case 4:		/*Update max task count*/
 #ifdef CONFIG_TASK_SCHED_HISTORY
 		update_maxtask_count(atoi(arg2));
 #else
 		dbg("To use this feature, kindly enable CONFIG_TASK_SCHED_HISTORY\n");
 #endif
-	} else if (strncmp(arg1, "max_irq_count", 13) == 0) {
+		break;
+	case 5:		/*Update maxirq_count*/
 #ifdef CONFIG_IRQ_SCHED_HISTORY
 		update_maxirq_count(atoi(arg2));
 #else
 		dbg("To use this feature, kindly enable CONFIG_IRQ_SCHED_HISTORY\n");
 #endif
-	} else if (strncmp(arg1, "max_sem_count", 13) == 0) {
+		break;
+	case 6:		/*Update maxsem_count*/
 #ifdef CONFIG_SEMAPHORE_HISTORY
 		update_maxsem_count(atoi(arg2));
 #else
 		dbg("To use this feature, kindly enable CONFIG_SEMAPHORE_HISTORY\n");
 #endif
-	} else if (strncmp("data_abort", arg1, 10) == 0) {
+		break;
+	case 7:		/*Simulate data abort*/
 		simulate_data_abort();
-	} else if (strncmp("prefetch_abort", arg1, 14) == 0) {
+		break;
+	case 8:		/*Simulate prefetch abort*/
 		simulate_prefetch_abort();
-	} else if (strncmp("undef_abort", arg1, 11) == 0) {
+		break;
+	case 9:		/*Simulate undefined abort*/
 		simulate_undef_abort();
-	} else if (strncmp("usr_assert", arg1, 10) == 0) {
+		break;
+	case 10:	/*Simulate user assert*/
 		simulate_user_assert();
-	} else if (strncmp("dump_stack", arg1, 10) == 0) {
+		break;
+	case 11:	/*Dumps current task stack*/
 #ifndef CONFIG_FRAME_POINTER
 		dbg("To use this feature, kindly enable CONFIG_FRAME_POINTER\n");
 #else
 		dump_stack();
 #endif
-	} else if (strncmp("dump_allstack", arg1, 13) == 0) {
+		break;
+	case 12:	/*Dumps stack of all the tasks*/
 #ifndef CONFIG_FRAME_POINTER
 		dbg("To use this feature, kindly enable CONFIG_FRAME_POINTER\n");
 #else
 		dump_all_stack();
 #endif
-	} else if (strncmp("help", arg1, 4) == 0) {
-		show_usage();
-	} else {
+		break;
+	default:
 		dbg("Invalid argument %s\n", arg1);
 		show_usage();
```

Comment:
It needs to return -1 by line 416, ret - 0 indicates success and -1 indicates failure.